api/client/build.go

```diff
@@ -104,12 +111,16 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	case urlutil.IsGitURL(specifiedContext) && hasGit:
 		tempDir, relDockerfile, err = getContextFromGitURL(specifiedContext, *dockerfileName)
 	case urlutil.IsURL(specifiedContext):
-		tempDir, relDockerfile, err = getContextFromURL(cli.out, specifiedContext, *dockerfileName)
+		tempDir, relDockerfile, err = getContextFromURL(outBuf, specifiedContext, *dockerfileName)
 	default:
 		contextDir, relDockerfile, err = getContextFromLocalDir(specifiedContext, *dockerfileName)
 	}
 
 	if err != nil {
+		if *suppressOutput {
+			fmt.Fprintln(cli.err, outBuf)
+			return fmt.Errorf("unable to prepare context: %s", err)
```

Comment:
I'd prefer to avoid the risk of someone updating the error text in one of these Errorf's but not the other, let's just let it fall thru.